### PR TITLE
Swift 3 migration 

### PR DESCRIPTION
**Pull request by vadymmarkov**
@zenangst @RamonGilabert @onmyway133 It was a bit painful migration, but it's ready now.

- [x] Swift 3
- [x] Public API changes
- [x] Code refactoring

### CODE REVIEW

File: Sources/Request/RequestCapsule.swift [hunk 1]
```diff
@@ -37,36 +37,36 @@ class RequestCapsule: NSObject, Requestable, NSCoding {
...
-      methodString = aDecoder.decodeObjectForKey(Key.Method.rawValue) as? String,
-      method = Method(rawValue: methodString),
-      resource = aDecoder.decodeObjectForKey(Key.Resource.rawValue) as? String,
-      parameters = aDecoder.decodeObjectForKey(Key.Parameters.rawValue) as? [String: AnyObject],
-      headers = aDecoder.decodeObjectForKey(Key.Headers.rawValue) as? [String: String],
-      etagPolicy = ETagPolicy(rawValue: aDecoder.decodeIntForKey(Key.EtagPolicy.rawValue)),
-      storePolicy = StorePolicy(rawValue: aDecoder.decodeIntForKey(Key.StorePolicy.rawValue)),
-      cachePolicy = NSURLRequestCachePolicy(rawValue: UInt(aDecoder.decodeIntForKey(Key.CachePolicy.rawValue)))
+      methodString = aDecoder.decodeObject(forKey: Key.method.rawValue) as? String,
+      let method = Method(rawValue: methodString),
```
Comment (onmyway133) [inline Sources/Request/RequestCapsule.swift]: Swift 3 requires different `let` now?

Comment (vadymmarkov) [inline Sources/Request/RequestCapsule.swift]: @onmyway133 Yes, and I'm not a big fan of this.

Comment (RamonGilabert) [inline Sources/Request/RequestCapsule.swift]: And you won't have where in conditionals either anymore. It kind of makes sense if you want one of them to be a `var`, though.

Comment (vadymmarkov) [inline Sources/Request/RequestCapsule.swift]: @RamonGilabert It was like that in Swift 2.

Comment (zenangst) [inline Sources/Request/RequestCapsule.swift]: Can you use `var` in optional unwrapping again? I thought that was removed in Swift 2.2 or something like that.

Comment (vadymmarkov) [inline Sources/Request/RequestCapsule.swift]: No, you can use `var` in Swift 2.2 as far as I remember.

Comment (zenangst) [inline Sources/Request/RequestCapsule.swift]: Oh, maybe I'm confusing it with function declarations then ^^

Comment (RamonGilabert) [inline Sources/Request/RequestCapsule.swift]: Can you do let, var let for instance in Swift 2, though, @vadymmarkov?

Comment (vadymmarkov) [inline Sources/Request/RequestCapsule.swift]:
```swift
if let a1 = b1, var a2 = b2 {
}
```

Comment (RamonGilabert) [inline Sources/Request/RequestCapsule.swift]: Then I don't understand.

**Pull request merged**